app/controllers/internal/page_redirects_controller.rb

```diff
@@ -0,0 +1,62 @@
+class Internal::PageRedirectsController < Internal::ApplicationController
+  layout "internal"
+
+  before_action :set_page_redirect, only: %i[edit update destroy]
+
+  def new
+    @page_redirect = PageRedirect.new
+  end
+
+  def create
+    @page_redirect = PageRedirect.new(new_page_redirect_params)
+
+    if @page_redirect.save
+      flash[:success] = "Page Redirect created successfully!"
+      redirect_to internal_page_redirects_path
+    else
+      flash[:danger] = @page_redirect.errors.full_messages.to_sentence
+      render new_internal_page_redirect_path
+    end
+  end
+
+  def index
+    @q = PageRedirect.order(created_at: :desc).ransack(params[:q])
+    @page_redirects = @q.result.page(params[:page] || 1).per(25)
+  end
+
+  def edit; end
+
+  def update
+    if @page_redirect.update(edit_page_redirect_params)
+      flash[:success] = "Page Redirect updated successfully!"
+      redirect_to edit_internal_page_redirect_path(@page_redirect)
+    else
+      flash[:danger] = @page_redirect.errors.full_messages.to_sentence
+      render :edit
+    end
+  end
+
+  def destroy
```

Comment:
Won't removing page redirects have an impact on SEO?

I wonder if this model should have soft delete, so instead of destroying redirects, we should `enable`/`disable` them

Reply:
That's a good question, it would. But it should be pretty rare that we would need to do this. Whether it's destroyed or disabled - the effect would be the same in terms of SEO. So it's more about the UX.

My original thinking was an `enabled/disabled` flag adds one more layer of complexity (a redirect can exist, but it's inactive). It could open up the mistake of an admin not realizing something was set incorrectly.

I could honestly go either way here. If we think it's better, I can add the flag in there.

Reply:
I agree with you, better not to have them in the DB, let's destroy! :D